### DISCUSSION TITLE
Search for performant Join

### DISCUSSION THREAD

**Discussion by chriz91g-vawi** [pocketbase/pocketbase#3979] [Q&A]
I am looking for the most efficient way to retrieve all tasks, of all projects a particular user has access to.

My first thought is to implement this by creating a database view, as I believe that the "expand" method is not suitable in this scenario. 

To provide some context, here is a quick overview of my database schema:

projects (id, titel, pusers (multiple relation)) 
tasks      (id, titel, project (single relation)) 

In the projects collection API Rules "@request.auth.id != "" && pusers.id ?= @request.auth.id"

**Comment by ganigeorgiev** [comment]
One possible solution could be to define a List API rule for the "tasks" collection like:

```js
@request.auth.id != "" &&
project.pusers.id ?= @request.auth.id
```

And then on the client side:

```js
// authenticate
await pb.collection("users").authWithPassword("test@example.com", "1234567890")

// retrieve a paginated list with all tasks the user has access to
const result = await pb.collection("tasks").getList(1, 20)
```
